Gems/DebugDraw/Code/Source/DebugDrawSystemComponent.cpp

```diff
@@ -995,7 +1020,38 @@ namespace DebugDraw
         AZ::TickRequestBus::BroadcastResult(newText.m_activateTime, &AZ::TickRequestBus::Events::GetTimeAtCurrentTick);
     }
 
-    void DebugDrawSystemComponent::CreateTextEntryForComponent(const AZ::EntityId& componentEntityId, const DebugDrawTextElement& element)
+    // carbonated begin : additional DebugDrawText methods
+#if defined(CARBONATED)
+    void DebugDrawSystemComponent::DrawScaledTextOnScreen(const AZStd::string& text, float fontScale, const AZ::Color& color, float duration)
+    {
+        AZStd::lock_guard<AZStd::mutex> locker(m_activeTextsMutex);
+        DebugDrawTextElement& newText = m_activeTexts.emplace_back();
+        newText.m_drawMode = DebugDrawTextElement::DrawMode::OnScreen;
+        newText.m_text = text;
+        newText.m_fontScale = fontScale > 0.0f ? fontScale : 1.0f;
+        newText.m_color = color;
+        newText.m_duration = duration;
+        AZ::TickRequestBus::BroadcastResult(newText.m_activateTime, &AZ::TickRequestBus::Events::GetTimeAtCurrentTick);
+    }
+
+    void DebugDrawSystemComponent::DrawScaledTextOnScreenPos(float x, float y, const AZStd::string& text, float fontScale, const AZ::Color& color, float duration, bool bCenter)
+    {
+        AZStd::lock_guard<AZStd::mutex> locker(m_activeTextsMutex);
+        DebugDrawTextElement& newText = m_activeTexts.emplace_back();
+        newText.m_drawMode = DebugDrawTextElement::DrawMode::OnScreen;
+        newText.m_text = text;
+        newText.m_fontScale = fontScale > 0.0f ? fontScale : 1.0f;
```

Comment:
It could be unexpected behavior, when we want to use zero scale  (here and in the line 1031) during font size animation

Reply:
Thanks, refactorted.